control-plane/Dockerfile

```diff
@@ -0,0 +1,152 @@
+# This Dockerfile contains multiple targets.
+# Use 'docker build --target=<name> .' to build one.
+#
+# Every target has a BIN_NAME argument that must be provided via --build-arg=BIN_NAME=<name>
+# when building.
+
+
+# ===================================
+# 
+#   Non-release images.
+#
+# ===================================
+
+# dev copies the binary from a local build
+# -----------------------------------
+# BIN_NAME is a requirement in the hashicorp docker github action 
+
+FROM alpine:3.15 AS dev
```

Comment:
@ishustava Dockerfile dev target is 90% the same as as default-release target. The only difference is where it copies the binary from. 

I am leaving some tech debt here that we should remove when we get rid of CircleCI. That debt is the local build places the built binary in /pkg/bin/<arch>/<binary> and this is the location that CircleCI depends on to do acceptance tests. When we fully move to Github Actions we can move everything to /dist